src/jit-diff/jit-diff.cs

```diff
@@ -159,25 +184,240 @@ private void SetRID()
+
+                if ((_outputPath == null) && (_diffRoot != null))
+                {
+                    _outputPath = Path.Combine(_diffRoot, s_defaultDiffDirectoryName);
+                    PathUtility.EnsureDirectoryExists(_outputPath);
+
+                    Console.WriteLine("Using --output={0}", _outputPath);
+                }
+
+                bool needTestTree = Benchmarks || DoTestTree;
+
+                if ((_platformPath == null) || (_basePath == null) || (_diffPath == null) || ((_testPath == null) && needTestTree))
+                {
+                    // Try all combinations of build and architecture to find one that has both a Core_Root
+                    // and, if necessary, base and diff builds and tests.
+                    //
+                    // If the user already specified one of --core_root, --base, or --diff, we leave
+                    // that alone, and only try to fill in the remaining ones with an appropriate
+                    // default.
+                    //
+                    // E.g.:
+                    //    test_root: c:\gh\coreclr\bin\tests\Windows_NT.x64.checked
+                    //    Core_Root: c:\gh\coreclr\bin\tests\Windows_NT.x64.checked\Tests\Core_Root
+                    //    base/diff: c:\gh\coreclr\bin\Product\Windows_NT.x64.checked
+
+                    List<string> archList;
+                    if (_arch == null)
                     {
-                        _moniker = _platformName;
+                        archList = new List<string> { "x64", "x86" };
                     }
-                    break;
-                    case "Darwin":
+                    else
                     {
-                        _moniker = "OSX";
+                        archList = new List<string> { _arch };
                     }
-                    break;
-                    default:
+
+                    List<string> buildList;
+                    if (_build == null)
                     {
-                        Console.WriteLine("No platform mapping!");
-                        _moniker = "bogus";
+                        buildList = new List<string> { "checked", "debug" };
+                    }
+                    else
+                    {
+                        buildList = new List<string> { _build };
+                    }
+
+                    bool found = false;
+                    foreach (var build in buildList)
+                    {
+                        foreach (var arch in archList)
+                        {
+                            var buildDirName = GetBuildOS() + "." + arch + "." + build;
+                            string tryPlatformPath = null, tryBasePath = null, tryDiffPath = null, tryTestPath = null;
+
+                            if ((_platformPath == null) && (_diffRoot != null))
+                            {
+                                tryPlatformPath = Path.Combine(_diffRoot, "bin", "tests", buildDirName, "Tests", "Core_Root");
```

Comment:
On Linux, I've always pointed `--core_root` to the overlay directory I created.  I think you've mentioned in the docs above that running `runtests.sh GenerateLayoutOnly` will populate this `Core_Root` with the right set of stuff that this will work on Linux, but wanted to make sure I'm getting that right -- is that what you're saying/expecting, that this formula works on all supported platforms assuming that the dev has done a `GenerateLayoutOnly` runtests?